easypwned/src/main.rs

```diff
@@ -1,131 +1,210 @@
+    }
 
-async fn handler_check(
-    Extension(bloom): Extension<Arc<EasyBloom>>,
-    extract::Json(payload): extract::Json<CheckRequestBody>,
-) -> Json<Value> {
-    let check = bloom.check(&payload.hash.as_bytes().to_vec());
-    Json(json!({
-        "hash": payload.hash,
-        "secure": !check,
-    }))
+    // todo: url ausm wasm_bindgen holen, da hab ich ja die url
+    Some(reqwest::get(format!("http://localhost:3342/pw/{:?}", password))
+        .await
+        .unwrap()
+        .json::<PwnedState>()
+        .await
+        .expect("Something went wrong"))
 }
 
-async fn handler_pw(
-    Extension(bloom): Extension<Arc<EasyBloom>>,
-    Path(pw): Path<String>,
-) -> Json<Value> {
-    let mut hasher = Sha1::new();
-    hasher.update(pw.as_bytes());
-    let hash_raw = &hasher.finalize();
-    let hash = base16ct::lower::encode_string(hash_raw).to_uppercase();
-
-    let check = bloom.check(&hash.as_bytes().to_vec());
-    Json(json!({
-        "pw": pw,
-        "hash": hash,
-        "secure": !check,
-    }))
+fn password_section(cx: Scope) -> Element {
+    let password = use_state(cx, || String::from(""));
+
+    let fetch_pwned_status = use_future(cx, (password.get(), ), |(password,)| async move {
+        fetch_pwned_state(password).await
+    });
+
+    let pwned_state = match fetch_pwned_status.value() {
+        Some(s) => s.clone(),
+        None => None,
+    };
+
+    let bg_color = match pwned_state.clone() {
+        Some(v) => if v.secure {
+            String::from("bg-emerald-400")
+        } else {
+            String::from("bg-red-400")
+        },
+        None => String::from("bg-sky-400")
+    };
+
+    info!("{:?}", pwned_state);
+
+    cx.render(rsx! {
+        div {
+            class: "{bg_color}",
+            section {
+                class: "mx-auto max-w-7xl py-6 sm:px-6 lg:px-8",
+                div {
+                    h1 {
+                        class: "text-3xl font-bold tracking-tight text-gray-900",
+                        "EasyPwned"
+                    }
+                    div {
+                        class: "my-3",
+                        "Would that the Argo had never winged its way to the land of Colchis through the dark-blue Symplegades!1 Would that the pine trees had never been felled in the glens of Mount Pelion and furnished oars for the hands of the heroes who at Pelias' command set forth in quest of the Golden Fleece! For then my lady Medea would not have sailed to the towers of Iolcus, her heart smitten with love for Jason, or persuaded the daughters of Pelias to kill their father and hence now be inhabiting this land of Corinth, <separated from her loved ones and country. At first, to be sure, she had, even in Corinth, a good life with her husband and children, an exile loved by the citizens to whose land she had come, and lending to Jason himself all her support. This it is that most rescues life from trouble, when a woman is not at variance with her husband."
```

Comment:
Example text, needs to be replaced with an actual explanation what is happening here